### PR TITLE
Java 21 and 2024.12.0 bump

### PR DESCRIPTION
**Pull request by gem-neo4j**
Update to Java 21 and Neo 2024.12.0

### CODE REVIEW

**Comment by vga91** [comment]
@gem-neo4j 

I don't know how far along this PR is, but I think this commit is also needed to make the CI green: https://github.com/neo4j-contrib/neo4j-apoc-procedures/pull/4225/commits/505ecd59e23ec549e1233947a9d15326bb3d2f05.

Otherwise, we'll get this error: https://github.com/neo4j-contrib/neo4j-apoc-procedures/actions/runs/11707717937/job/32716503891?pr=4222#step:10:1855.

Or at least, I had this error before the Java 21-related errors, I guess it's still present..

**Comment by gem-neo4j** [comment]
Ah thanks! will take a look :)

**Comment by vga91** [comment]
@gem-neo4j 
To solve CI error, I guess we have to change [here](https://github.com/neo4j/apoc/blob/dev/core/src/main/java/apoc/help/Help.java#L48), 
to handle "extendedCypher25.txt" and "extendedCypher5.txt" instead of / besides "extended.txt" :)

**Comment by gem-neo4j** [comment]
@vga91 yep, I have another PR for that on core :) (was out sick so sorry for late reply), will try to get it all working this week!

File: extended/src/main/java/apoc/load/LoadJsonExtended.java [hunk 1]
```diff
@@ -46,8 +46,7 @@ public class LoadJsonExtended {
     public URLAccessChecker urlAccessChecker;
```
Comment (vga91) [inline extended/src/main/java/apoc/load/LoadJsonExtended.java:46]: `LoadJsonExtended` class needs to have `@Extended`

**Pull request merged**